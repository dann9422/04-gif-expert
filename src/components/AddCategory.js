import React, { useState } from 'react'
import PropTypes from 'prop-types';
export const AddCategory = ({setcategories}) => {

    const [inputValue, setinputValue] = useState('')
    const handleInpuntChange = (e)=>{

        setinputValue(e.target.value);
        console.log('handleInpuntChange llamdo');
    }

    const handleSubmit =(e)=>{
e.preventDefault();
// console.log('enviado desde el formulario ')
if(inputValue.trim().length>2){
    setcategories(cats=>[inputValue,...cats]);
setinputValue('');
}


}
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input type="text" placeholder='Ingrese la categoria que desea buscar'
            value={inputValue}
            onChange={handleInpuntChange}/>
            
           
        </form>
    )
}


AddCategory.prototype={
    setcategories: PropTypes.func.isRequired

}