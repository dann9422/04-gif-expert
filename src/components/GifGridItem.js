import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({id,title,url}) => {
    // console.log(id,title,url)
    return (
        <div className='card animate__animated animate__lightSpeedInRight'>
           <img src={url} alt={title}></img>
           <p>{title}</p>
        </div>
    )
}


GifGridItem.prototype={
    title: PropTypes.string.isRequiered,
    url : PropTypes.string.isRequired

}

