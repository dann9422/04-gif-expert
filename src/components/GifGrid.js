import React, { useEffect, useState } from 'react'
import { getgif } from '../helppers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {


    const [images, setimages] = useState([]);
        useEffect(()=>{
            getgif(category)
            .then(setimages)
        },[category])

    
        


    return (
        <>
        <h3>{category}</h3>
        <div className='card-grid'>
                {
                    images.map(img=>(
                        <GifGridItem
                        key={img.id}
                        {...img }
                        
                        />
                    ))
                }
            
        </div>
        </>
    )
}
