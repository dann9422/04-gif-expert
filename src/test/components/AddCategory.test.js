
import { shallow } from "enzyme";
import React from "react";
import { AddCategory } from "../../components/AddCategory";



describe('pruebas en el componente', () => {
    const setcategories= ()=>{}
    const  wrappeer = shallow(<AddCategory setcategories={setcategories}/>);
  test('debe de mostrarse correctamente ', () => {
    
    expect(wrappeer).toMatchSnapshot();

  });
  test('evaluar los cambios en el input de texto', () => {
    const input =wrappeer.find('input');
    const value ='hola mundo'
    input.simulate('change',{target:{value}});
    expect(wrappeer.find('p').text().trim()).toBe(value);
  });
  
  
});
