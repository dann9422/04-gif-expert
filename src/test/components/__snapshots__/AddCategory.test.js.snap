// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente debe de mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Ingrese la categoria que desea buscar"
    type="text"
    value=""
  />
</form>
`;
